src/eventbridge/target.ts

```diff
@@ -0,0 +1,53 @@
+import { aws_events, aws_events_targets } from "aws-cdk-lib";
+import { EventBus } from "./bus";
+import { Function, isFunction } from "../function";
+import { EventBusRuleInput } from "./types";
+
+export type LambdaTargetProps<P> = {
+  func: Function<P, any>;
+} & Omit<aws_events_targets.LambdaFunctionProps, "event">;
+
+export type EventBusTargetProps<P extends EventBusRuleInput> = {
+  bus: EventBus<P>;
+} & aws_events_targets.EventBusProps;
+
+export type EventBusTargetResource<T extends EventBusRuleInput, P> =
+  | Function<P, any>
+  | LambdaTargetProps<P>
+  | EventBus<T>
+  | EventBusTargetProps<T>;
+
+  /**
+   * Add a target to the run based on the configuration given.
+   */
+export const pipe = <T extends EventBusRuleInput, P>(
```

Comment:
any reason why this isn't in the `rule.ts`? Is it used anywhere else?

Reply:
Yeah, `pipe` is used in `rule.ts` and `transform.ts` because we can `pipe` directly from a rule (`.when().pipe()`, no transform) or after a transform (`.when().map().pipe()`, with transform). `pipe` generalizes the logic between the two, including failing if there is an event target input (transform), but the target is event bus which doesn't support transforming.